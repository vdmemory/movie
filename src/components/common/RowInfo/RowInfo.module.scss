.rowInfo {
    display: flex;
    align-items: center;
    padding: 5px 0;
    & .title {
        margin: 0;
        font-size: 20px;
    }

    & .subtitle {
        margin: 0 0 0 15px;
        font-size: 18px;
    }
}