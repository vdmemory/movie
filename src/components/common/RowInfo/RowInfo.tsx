import React from "react";

import styles from "./RowInfo.module.scss";

type Props = {
    title: string;
    subtitle: string;
};

const RowInfo: React.FC<Props> = ({ title, subtitle }) => {
    return (
        <div className={styles.rowInfo}>
            <h3 className={styles.title}>{title}</h3>
            <p className={styles.subtitle}>{subtitle}</p>
        </div>
    );
};

export default RowInfo;
