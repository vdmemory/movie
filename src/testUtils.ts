export const deepEqual = (a: any, b: any) => {
    if (a === b) {
        return true;
    }

    if (
        a == null ||
        typeof a != "object" ||
        b == null ||
        typeof b != "object"
    ) {
        return false;
    }

    var propertiesInA = 0,
        propertiesInB = 0;
    for (var property in a) {
        propertiesInA += 1;
    }
    for (var property in b) {
        propertiesInB += 1;
        if (!(property in a) || !deepEqual(a[property], b[property])) {
            return false;
        }
    }
    return propertiesInA == propertiesInB;
};
