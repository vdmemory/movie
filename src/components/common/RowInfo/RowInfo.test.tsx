import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import RowInfo from "./RowInfo";
import { deepEqual } from "../../../testUtils";

type Props = {
  title: string;
  subtitle: string;
};

const setup = (props: Props) => {
  const renderer = createRenderer();
  renderer.render(<RowInfo {...props} />);
  const output = renderer.getRenderOutput();
  return { output, props };
};

const propsRender: Props = {
  title: "Title",
  subtitle: "Subtitle",
};

const propsNotRender: Props = {
  title: "Title",
  subtitle: "N/A",
};

describe("RowInfo Component", () => {
  describe("Render component and its components", () => {
    test("Should renders correctly", () => {
      const { output } = setup(propsRender);
      expect(output).toMatchSnapshot();
    });

    test("Should render subtitle class and text title", () => {
        const { output } = setup(propsRender);
        const [title] = output.props.children;
        expect(title.props.className).toBe("title");
        expect(title.props.children).toBe("Title: ");
    });

    test("Should render subtitle class and text Subtitle", () => {
        const { output } = setup(propsRender);
        const [, subtitle] = output.props.children;
        expect(subtitle.props.className).toBe("subtitle");
        expect(subtitle.props.children).toBe("Subtitle");
    });

    describe("Check props", () => {
        const expectedProps = {
            title: "Title",
            subtitle: "Subtitle",
        };
        test("Should render and check props", () => {
            const { props } = setup(propsRender);
            expect(deepEqual(props, expectedProps)).toBeTruthy();
        });
    });
  });

  describe("When a component is not rendered", () => {
    test("Should not render the component when the subtitle is 'N/A'", () => {
      const { output } = setup(propsNotRender);
      expect(output).toBe(null);
    });
  });
});
