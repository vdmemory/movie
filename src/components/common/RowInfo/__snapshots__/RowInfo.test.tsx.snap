// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowInfo Component Render component and its components Should renders correctly 1`] = `
<div
  className="rowInfo"
>
  <h3
    className="title"
  >
    Title: 
  </h3>
  <p
    className="subtitle"
  >
    Subtitle
  </p>
</div>
`;
